### PR TITLE
Feat/routes page

### PR DESCRIPTION
**Pull request by joaoqueiroz95**
Route Selection screen

image

Added unit tests as well

### CODE REVIEW

**Review by mathvaillant — APPROVED** [review]
It is better to use [the built in React Native flatlist](https://reactnative.dev/docs/flatlist) instead of doing maps. But as the maps were only applied to screens which obviously won't have 1000 list items, I don't think it's an issue at all. Keep in mind that if you have a big list, use Flatlist, never map. 

Besides that, looks good to me!

File: __tests__/RouteSelection.test.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      { name: "route 2", destination: { code: "", name: "" }, origin: { code: "", name: "" } },
+    ]);
+
+    const { getAllByTestId, getByText } = render(<RouteSelectionScreen />);
+
+    expect(getRoutes).toHaveBeenCalled();
+
+    await waitFor(() => {
+      expect(getByText("route 1")).toBeTruthy();
+      expect(getByText("route 2")).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (joaoqueiroz95) [reply __tests__/RouteSelection.test.tsx:34]: Why would it fail?
The /routes request returned "route 1" and "route 2". So it should appear 2 buttons with these names

**Comment by joaoqueiroz95** [comment]
> It is better to use [the built in React Native flatlist](https://reactnative.dev/docs/flatlist) instead of doing maps. But as the maps were only applied to screens which obviously won't have 1000 list items, I don't think it's an issue at all. Keep in mind that if you have a big list, use Flatlist, never map.
> 
> Besides that, looks good to me!

Thanks for informing about FlatList, did not know it.
I searched a bit, and FlatList adds a bit of overhead, so for small lists I think map is fine. I believe this list will be small.

File: __tests__/RouteSelection.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (joel-gouveia) [reply __tests__/RouteSelection.test.tsx:34]: Tomorrow the names are route 3 and route 4 what happens then?

[the remark this answers is not in the document]
Comment (joaoqueiroz95) [reply __tests__/RouteSelection.test.tsx:34]: But here I am mocking to return route 1 and route 2, I am not sure I understand the problem

File: src/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+export function Footer({ buttons = [] }: IFooter) {
+  return (
+    <View style={styles.container}>
+      {buttons.map(({ label, onPress }) => (
+        <View key={label} style={styles.btnContainer}>
+          <Button onPress={onPress} variant="outline" style={styles.button} testID="footer-btn ">
+            {label}
+          </Button>
+        </View>
+      ))}
```
Comment (joel-gouveia) [inline src/components/Footer/Footer.tsx:18]: For readability make this a smaller component or add it to a function

Comment (joaoqueiroz95) [reply src/components/Footer/Footer.tsx]: Just updated src/components/Footer/Footer.tsx, let me know if it is good now

**Comment by joel-gouveia** [comment]
> It is better to use [the built in React Native flatlist](https://reactnative.dev/docs/flatlist) instead of doing maps. But as the maps were only applied to screens which obviously won't have 1000 list items, I don't think it's an issue at all. Keep in mind that if you have a big list, use Flatlist, never map.
> 
> Besides that, looks good to me!

Expanding on this, it is a good point.. If in terms of performance we are safeguarding the future lets do this now. It is the moment to structure and think ahead

File: src/context/auth.tsx [hunk 3]
```diff
@@ -1,24 +1,36 @@
...
 }
 
 export const AuthContext = createContext({} as AuthContextProps);
 
 export function AuthContextProvider({ children }: { children: ReactNode }) {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const [isAuthenticating, setIsAuthenticating] = useState(true);
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const [isAuthenticated, setIsAuthenticated] = useState(true);
+  const [isAuthenticating, setIsAuthenticating] = useState(false);
```
[the remark this answers is not in the document]
Comment (joaoqueiroz95) [reply src/context/auth.tsx]: Fixed

**Comment by joaoqueiroz95** [comment]
> > It is better to use [the built in React Native flatlist](https://reactnative.dev/docs/flatlist) instead of doing maps. But as the maps were only applied to screens which obviously won't have 1000 list items, I don't think it's an issue at all. Keep in mind that if you have a big list, use Flatlist, never map.
> > Besides that, looks good to me!
> 
> Expanding on this, it is a good point.. If in terms of performance we are safeguarding the future lets do this now. It is the moment to structure and think ahead.
> 
> Thank you for flagging this @mathvaillant

Just substituted it

**Review by joel-gouveia — APPROVED** [review]
LGTM!

**Pull request merged**